### PR TITLE
Make the deploy script more general (test and production) and add a test deployment to the github workflow

### PR DESCRIPTION
**Pull request by theref**

### CODE REVIEW

File: scripts/deploy_subscription_manager.py [hunk 1]
```diff
@@ -1,21 +1,37 @@
 #!/usr/bin/python3
 
+from multiprocessing.sharedctypes import Value
```
Comment (cygnusv) [inline scripts/deploy_subscription_manager.py:3]: This is probably a left-over?

File: scripts/deploy_subscription_manager.py [hunk 2]
```diff
@@ -25,10 +41,10 @@ def main(account_name : str = 'test'):
...
-        proxy_admin.address,
-        calldata,
-        {'from': deployer})
+        subscription_manager_logic.address, proxy_admin.address, calldata, {"from": deployer}
+    )
 
-    subscription_manager = Contract.from_abi("SubscriptionManager", transparent_proxy.address, subscription_manager_logic.abi, owner=None)
+    subscription_manager = Contract.from_abi(
+        "SubscriptionManager", transparent_proxy.address, subscription_manager_logic.abi, owner=None
+    )
```
Comment (cygnusv) [inline scripts/deploy_subscription_manager.py:49]: I merged a few PRs today, so this part should go after rebasing.

**Review by cygnusv — APPROVED** [review]
Great work, just needs rebasing!

File: scripts/deploy_subscription_manager.py (hunk 1 above)
Comment (theref) [reply scripts/deploy_subscription_manager.py]: done

File: scripts/deploy_subscription_manager.py (hunk 2 above)
Comment (theref) [reply scripts/deploy_subscription_manager.py]: rebased

**Pull request merged**